shell/platform/android/io/flutter/view/FlutterNativeView.java

```diff
@@ -128,7 +128,14 @@ private void attach(FlutterNativeView view) {
 
     // Called by native to send us a platform message.
     private void handlePlatformMessage(final String channel, byte[] message, final int replyId) {
-        assertAttached();
+        // The platform may not be attached immediately in certain cases where a new bundle is run -
+        // the native view is created in a separate thread. This mostly happens when the app restarts in dev
+        // mode when switching into split-screen mode. Preventing app restarts on layout and density
+        // changes will prevent this, and afterwards this can be changed back to an assert.
+        if (!isAttached()) {
```

Comment:
Does this still happen if you apply the pending engine refactoring by @chinmaygarde ? (https://github.com/chinmaygarde/flutter_engine/tree/shell)

The Dart application may fail if it sends a message to the host and the host drops the message with no response.

Reply:
So looking into this more - I can reproduce the issue on master, somewhat flaky though.  Might only be a debug issue